### PR TITLE
#80 Refresh from upstream

### PR DESCRIPTION
**Pull request by Dan7-7-7**
https://github.com/Skyscanner/turbolift/issues/80

### CODE REVIEW

File: cmd/clone/clone.go [hunk 1]
```diff
@@ -113,6 +113,27 @@ func run(c *cobra.Command, _ []string) {
 			continue
 		}
 		createBranchActivity.EndWithSuccess()
+
+		if nofork {
```
Comment (sledigabel) [inline cmd/clone/clone.go:117]: i got confused with this for a minute.
Wouldn't it be clearer if you put the block below in a `if !nofork {` statement?

Comment (Dan7-7-7) [reply cmd/clone/clone.go:117]: Yeah can do, I somehow thought it was neater to reduce the amount of logic within the conditional block. But see your point - will do it that way instead.

**Comment by rnorth** [comment]
I think this looks promising - haven't had a chance to try it locally yet but it's looking good 👍

File: internal/github/github.go [hunk 2]
```diff
@@ -87,6 +88,11 @@ func (r *RealGitHub) ClosePullRequest(output io.Writer, workingDir string, branc
 	return execInstance.Execute(output, workingDir, "gh", "pr", "close", fmt.Sprint(pr.Number))
 }
 
+func (r *RealGitHub) GetDefaultBranchName(output io.Writer, workingDir string, fullRepoName string) (string, error) {
+	defaultBranch, err := execInstance.ExecuteAndCapture(output, workingDir, "gh", "repo", "view", fullRepoName, "--json", "defaultBranchRef", "--jq", ".defaultBranchRef.name")
+	return strings.Trim(defaultBranch, "\n"), err
```
Comment (Dan7-7-7) [inline internal/github/github.go:93]: the command returns the branch name with a \n appended. Tried a few other options, e.g. 
`gh repo view bd/reviewstruck --json defaultBranchRef --template {{.defaultBranchRef.name}}`
`gh api repos/bd/reviewstruck --template {{.default_branch}}`
but the same happens.

File: internal/github/fake_github.go [hunk 3]
```diff
@@ -76,33 +82,54 @@ func NewFakeGitHub(h func(output io.Writer, workingDir string, fullRepoName stri
...
+		}
+		return true, nil
+	}, func(output io.Writer, workingDir string) (interface{}, error) {
+		return PrStatus{}, nil
+	})
+}
+
+type Command int
+
+const (
```
Comment (sledigabel) [inline internal/github/fake_github.go:129]: if you don't particularly need the int value, you can use `iota`.

**Review by sledigabel — APPROVED** [review]
LGTM, adds an auto fetch to the upstream main branch in case of a fork.

**Review by rnorth — APPROVED** [review]
LGTM - thanks @Dan7-7-7!

**Review by sledigabel — APPROVED** [review]
LGTM

**Pull request merged**